spec/factories/work_factory.rb

```diff
@@ -13,12 +13,13 @@
     # Automatically set a representative if needed and possible
     after(:build) do |work|
       if work.representative.nil? && work.members.loaded?
-        work.representative = work.members.first
+        work.representative = work.members.to_a.find {|w| w.published? }
       end
     end
 
     trait :published do
       published { true }
+      members {  [ build(:asset_with_faked_file, published: true) ] } # published works need a representative now
```

Comment:
asset_with_faked_file creates a `published: true` Asset already. It's fine if you want to `published: true` there to make it even more clear/explicit, maybe that's even best? I just mention it because I see you adding `published: true` every time you are using the factory, and I wasn't sure if you realized that was not necessary?

Reply:
I didn't give it much thought until this week but now of course I'm hyper-aware of it!